views/index.mst

```diff
@@ -57,94 +45,107 @@ html,body {
         <p class="finalTime"></p>
         <p class="finalLocation">{{#finalLocations}}<span>{{.}}</span>{{/finalLocations}}</p>
     </div>
     {{/finalsList}}
 </div>
 <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/moment.js/2.22.2/moment.js"></script>
 <script type="text/javascript">
-    window.onload = function() {
-        const times = [];
-        const now = moment();
+window.onload = function() {
+    const times = [];
+    const now = moment();
 
-        Array.from(document.getElementsByClassName("finalTime")).map((ft, i) => {
+    Array.from(document.getElementsByClassName("finalTime")).map((ft, i) => {
             const final = ft.parentNode;
             const start = moment(final.dataset.finalStart);
             const end = moment(final.dataset.finalEnd);
             const past = (end.dayOfYear() < now.dayOfYear());
             times[i] = {start, end, past};
-            ft.innerHTML = start.format("ddd MMM DD, h:mm A");
+            ft.innerHTML = start.format("ddd MMM DD, hh:mm A");
             if (past) {
                 final.classList.add("past");
             }
-        });
-
-        function updateCountdowns() {
-            const now = moment();
-            Array.from(document.getElementsByClassName("finalCountdown")).map((fc, i) => {
+            });
+    function updateCountdowns() {
+        const now = moment();
+        Array.from(document.getElementsByClassName("finalCountdown")).map((fc, i) => {
                 if (times[i].past) return;
                 const diff = moment.duration(times[i].start.diff(now));
-                const hrs = Math.floor(diff.asHours());
-                if (hrs < 0) {
-                    fc.parentElement.classList.add("countdownExpired");
-                    times[i].past = (times[i].end.dayOfYear() < now.dayOfYear());
-                    fc.innerHTML = "00:00:00";
+                const _hrs = Math.floor(diff.asHours());
+                if (_hrs < 0) {
+                    fc.parentElement.classList.remove("countdownWarning");
+                    if (times[i].start.unix() < now.unix() && times[i].end.unix() > now.unix()) {
+                        fc.parentElement.classList.add("inProgress")
+                        fc.innerHTML = "ongoing"
+                    }
+                    if (times[i].end.unix() < now.unix()) {
+                        fc.parentElement.classList.remove("inProgress");
+                        fc.parentElement.classList.add("past");
+                        fc.parentElement.classList.add("countdownExpired");
+                        times[i].past = (times[i].end.dayOfYear() < now.dayOfYear());
+                        fc.innerHTML = "00:00:00";
+                    }
                     return;
                 }
+                if (_hrs < 2) {
+                    fc.parentElement.classList.add("countdownWarning")
+                }
+                
+                let hrs = diff.hours();
```

Comment:
This is incorrect. .hours() returns the number of hours modulo 24 (ie subtracting off full days). Needs to be .asHours()

Reply:
.asHours() breaks it. it works with .hours()
![image](https://user-images.githubusercontent.com/5100938/49875270-3519f880-fde6-11e8-9d50-1cc82c16d4f9.png)
with asHours()
![image](https://user-images.githubusercontent.com/5100938/49875314-54188a80-fde6-11e8-9ccb-dfbf51b305b1.png)
with hours()

Reply:
I can't paste a picture on mobile, but look at Thursday and Friday's classes. The number of hours is wrong.

Reply:
.asHours returns a float, you need to round it down

Reply:
fixed